### PR TITLE
Llama Padding-Free for Finetune

### PR DESCRIPTION
**Pull request by fabianlim**
@garrett361 This is the first PR to merge back to `main`
- support padding-free for `finetune.py`
- suggest we revert all the formatting changes that are not necesary

### CODE REVIEW

File: open_instruct/finetune.py [hunk 1]
```diff
@@ -748,10 +798,21 @@ def main(args: FlatArguments):
         logger.info(f"Sample {index} of the training set: {train_dataset[index]}.")
 
     # DataLoaders creation:
+    if args.padding_free:
+        accelerator.print("Using padding-free collation")
```
Comment (fabianlim) [inline open_instruct/finetune.py:802]: are these `accelerator.print` statements needed @garrett361 ?

Comment (garrett361) [reply open_instruct/finetune.py:802]: No, just for testing

**Comment by fabianlim** [comment]
this PR has been depracated in favour of #12
